gpMgmt/bin/gppylib/operations/test/unit/test_unit_backup_utils.py

```
@@ -841,28 +856,52 @@ def test_get_full_timestamp_for_incremental_with_prefix_bad_files(self, mock1, m
         self.context.dump_prefix = 'foo_'
         with self.assertRaisesRegexp(Exception, 'No full backup found for incremental'):
             get_latest_full_dump_timestamp(self.context)
 
-    def test_convert_report_filename_to_cdatabase_filename_with_prefix_default(self):
-        report_file = '/data/db_dumps/20160101/bar_gp_dump_20160101010101.rpt'
-        expected_output = '/data/db_dumps/20160101/bar_gp_cdatabase_1_1_20160101010101'
+    @patch('gppylib.operations.backup_utils.get_lines_from_file', return_value=['BackupFile /data/master/db_dumps/20160101/bar_gp_dump_1_1_20160101010101.gz: Succeeded'])
+    def test_convert_report_filename_to_cdatabase_filename_with_prefix_old_format(self, mock):
+        report_file = '/data/master/db_dumps/20160101/bar_gp_dump_20160101010101.rpt'
+        expected_output = '/data/master/db_dumps/20160101/bar_gp_cdatabase_1_1_20160101010101'
         self.context.dump_prefix = 'bar_'
         cdatabase_file = convert_report_filename_to_cdatabase_filename(self.context, report_file)
         self.assertEquals(expected_output, cdatabase_file)
 
+    @patch('gppylib.operations.backup_utils.get_lines_from_file', return_value=['BackupFile /data/master/db_dumps/20160101/bar_gp_dump_-1_1_20160101010101.gz: Succeeded'])
+    def test_convert_report_filename_to_cdatabase_filename_with_prefix_new_format(self, mock):
+        report_file = '/data/master/db_dumps/20160101/bar_gp_dump_20160101010101.rpt'
+        expected_output = '/data/master/db_dumps/20160101/bar_gp_cdatabase_-1_1_20160101010101'
+        self.context.dump_prefix = 'bar_'
+        cdatabase_file = convert_report_filename_to_cdatabase_filename(self.context, report_file)
+        self.assertEquals(expected_output, cdatabase_file)
+
+    @patch('gppylib.operations.backup_utils.get_lines_from_file', return_value=[''])
+    def test_convert_report_filename_to_cdatabase_filename_with_prefix_empty_file(self, mock):
```

Why is it okay for this function to return a converted file name, if the file is empty?
If the file is empty, doesn't that mean that the dump failed?
Should we raise an exception in the function if the contents are empty?